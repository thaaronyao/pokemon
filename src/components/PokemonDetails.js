import React from 'react';
import PokemonDisplay from './PokemonDisplay';

const PokemonDetails = ({pokemon, resetPokemonNum, getPokemonNum}) => {
    return(
        <div >
            <div className="details">
                {/* <h2>{pokemon.name}</h2> */}
                <table>
                    <caption>{pokemon.name}</caption>
                    <tr>
                        <th>Max HP</th>
                        <th>Max CP</th>
                        <th>number</th>
                        <th>types</th>
                        <th>id</th>
                    </tr>
                    <tr>
                        <td>{pokemon.maxHP}</td>
                        <td>{pokemon.maxCP}</td>
                        <td>{pokemon.number}</td>
                        <td>{pokemon.types.join(' ')}</td>
                        <td>{pokemon.id}</td>
                    </tr>
                </table>
                <img src={pokemon.image} />
            </div>
            <div className="pokemons">
                <p>Evolution: </p>
                {pokemon.evolutions? 
                pokemon.evolutions.map(pokemon => (
                    <PokemonDisplay 
                    getPokemonNum={getPokemonNum} 
                    pokemon={pokemon}/>
                )): <p>No More Evolution</p>}
            </div>
            <button onClick={resetPokemonNum}>Go Back to Home</button>
        </div>
    )
}

export default PokemonDetails;