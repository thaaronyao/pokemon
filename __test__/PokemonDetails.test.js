import React from 'react';
import renderer from 'react-test-renderer';

import PokemonDetails from '../src/components/PokemonDetails';

test('render test', () => {
    const props = {   
            image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
            name: "Bulbasaur",
            maxCP: 951,
            maxHP: 1071,
            number: "001",
            id: "UG9rZW1vbjowMDE=",
            types: ["Grass", "Poison"],
            evolutions: []
        }
    const component = renderer.create(<PokemonDetails pokemon={props}/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})