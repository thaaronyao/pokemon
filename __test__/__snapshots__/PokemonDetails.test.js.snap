// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<div>
  <div
    className="details"
  >
    <table>
      <caption>
        Bulbasaur
      </caption>
      <tr>
        <th>
          Max HP
        </th>
        <th>
          Max CP
        </th>
        <th>
          number
        </th>
        <th>
          types
        </th>
        <th>
          id
        </th>
      </tr>
      <tr>
        <td>
          1071
        </td>
        <td>
          951
        </td>
        <td>
          001
        </td>
        <td>
          Grass Poison
        </td>
        <td>
          UG9rZW1vbjowMDE=
        </td>
      </tr>
    </table>
    <img
      src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
    />
  </div>
  <div
    className="pokemons"
  >
    <p>
      Evolution: 
    </p>
  </div>
  <button>
    Go Back to Home
  </button>
</div>
`;
